src/state/actions/dispatchArIOInteraction.ts

```diff
@@ -202,60 +199,37 @@ export default async function dispatchArIOInteraction({
         }
         break;
       case ARNS_INTERACTION_TYPES.PRIMARY_NAME_REQUEST: {
+        await arioContract.setPrimaryName(
+          {
+            name: payload.name,
+          },
+          {
+            ...WRITE_OPTIONS,
+            onSigningProgress: (step, payload) => {
+              if (step === 'requesting-primary-name') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Requesting primary name '${payload.name}'`,
+                });
+              } else if (step === 'request-already-exists') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Primary name request for '${payload.name}' already exists!`,
+                });
+              } else if (step === 'approving-request') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Approving primary name request for '${payload.name}'`,
+                });
+              }
+            },
+          },
+        );
+        // send a final confirmation message
         dispatch({
           type: 'setSigningMessage',
-          payload: 'Confirming Primary Name 1/2',
-        });
-        const existingPrimaryNameRequest = await arioContract
-          .getPrimaryNameRequest({
-            initiator: owner.toString(),
-          })
-          .catch((e) => {
-            console.error(e);
-            return undefined;
-          });
-
-        if (
-          !existingPrimaryNameRequest ||
-          existingPrimaryNameRequest.name !== payload.name
-        ) {
-          await arioContract
-            .requestPrimaryName(
-              {
-                name: payload.name,
-                fundFrom: originalFundFrom,
-                referrer: APP_NAME,
-              },
-              WRITE_OPTIONS,
-            )
-            .catch((e) => {
-              throw new Error('Unable to request Primary name: ' + e.message);
-            });
-        }
-        // UX sleep between transactions
-        await sleep(2000);
-
-        const antProcess = ANT.init({
-          hyperbeamUrl: hyperbeamUrl,
-          signer,
-          // we're not using hyperbeam here as we're writing to the contract
-          process: new AOProcess({
-            ao: antAo,
-            processId: payload.antProcessId,
-          }),
-        });
-
-        dispatch({
-          type: 'setSigningMessage',
-          payload: 'Confirming Primary Name 2/2',
-        });
-        await sleep(2000);
-        result = await antProcess.approvePrimaryNameRequest({
-          name: payload.name,
-          address: owner.toString(),
-          arioProcessId: payload.arioProcessId,
+          payload: `Successfully set primary name '${payload.name}'`,
```

Comment:
i dont see where we are setting the result here

Reply:
ops, didn't push - added here [6338dd1](https://github.com/ar-io/arns-react/pull/848/commits/6338dd1b28c458ecec9e22bcfe09c1b5eb31ee38)